### PR TITLE
Ft exercise #157110841

### PR DESCRIPTION
**Pull request by FelixWambiri**
**What does this PR do?**
* Show all infos for an exercise

**Description of Task to be completed?**
* Create a new read-only API endpoint that collects all the information of an exercise such as muscles, category, images, etc.

**How should this be manually tested?**
* After cloning the repo,cd into it and start the development server:
use the address provided after you run the server together with `/api/v2/exercisedetails/<id>` endpoint to get all the info of the exercise which has the id that you have provided

**What are the relevant pivotal tracker stories?**
* ID : #157110841

### CODE REVIEW

File: wger/exercises/tests/test_exercise.py [hunk 1]
```diff
@@ -403,6 +403,48 @@ class DeleteExercisesTestCase(WorkoutManagerDeleteTestCase):
     user_success = 'admin'
     user_fail = 'test'
 
+class ExerciseDetailsAPITestCase(WorkoutManagerTestCase):
+    """
```
Comment (betsybeth) [inline wger/exercises/tests/test_exercise.py:407]: Good implementation on writing docstrings

File: wger/exercises/tests/test_exercise.py [hunk 2]
```diff
@@ -403,6 +403,48 @@ class DeleteExercisesTestCase(WorkoutManagerDeleteTestCase):
...
+    Tests the exercise details are returned
+    """
+
+    def test_exercise_details(self, editor=False):
+        """
```
Comment (betsybeth) [inline wger/exercises/tests/test_exercise.py:412]: Good implementation on testing

File: wger/exercises/tests/test_exercise.py [hunk 3]
```diff
@@ -403,6 +403,48 @@ class DeleteExercisesTestCase(WorkoutManagerDeleteTestCase):
...
+        self.assertEqual(result['name_original'], '')
+        self.assertContains(response, 'creation_date')
+        self.assertContains(response, 'uuid')
+        self.assertEqual(result['license'], 2)
+        self.assertEqual(result['category'], 2)
+        self.assertEqual(result['language'], 2)
+        self.assertEqual(result['muscles'][0]['id'], 2)
+        self.assertContains(response, 'muscles_secondary')
+        self.assertEqual(result['equipment'], [])
+
```
Comment (Kalela) [inline wger/exercises/tests/test_exercise.py]: Great work testing broadly

File: wger/exercises/api/views.py [hunk 4]
```diff
@@ -207,3 +208,17 @@ class MuscleViewSet(viewsets.ReadOnlyModelViewSet):
     ordering_fields = '__all__'
     filter_fields = ('name',
                      'is_front')
+
+class ExerciseDetailsViewSet(viewsets.ReadOnlyModelViewSet):
+    '''
+    This is a read only API endpoint for overall exercise details 
+    '''
+    serializer_class = ExerciseDetailsSerializer
```
Comment (meshnesh) [inline wger/exercises/api/views.py]: This variable has been defined, but never used, is it apart of the code?

File: wger/exercises/api/views.py [hunk 5]
```diff
@@ -207,3 +208,17 @@ class MuscleViewSet(viewsets.ReadOnlyModelViewSet):
...
+
+    def get_queryset(self):
```
Comment (meshnesh) [inline wger/exercises/api/views.py]: Add a doctoring explaining the function

File: wger/exercises/tests/test_exercise.py [hunk 6]
```diff
@@ -512,6 +554,8 @@ def test_canonical_form_cache_delete(self):
             self.assertFalse(cache.get(cache_mapper.get_workout_canonical(workout_id)))
 
 
+
+
 # TODO: fix test, all registered users can upload exercises
```
Comment (meshnesh) [inline wger/exercises/tests/test_exercise.py:559]: remove extra commented code if not being used

**Review by meshnesh — APPROVED** [review]
Nice implementation of the feature.

File: wger/exercises/api/serializers.py [hunk 7]
```diff
@@ -71,3 +73,27 @@ class MuscleSerializer(serializers.ModelSerializer):
     '''
     class Meta:
         model = Muscle
+
+
+# Create a class serializer for all the details and information related to an exercise and all the fields are read only
```
Comment (frankip) [inline wger/exercises/api/serializers.py]: I like the way you have documented your code good job

File: wger/exercises/api/views.py [hunk 8]
```diff
@@ -207,3 +208,17 @@ class MuscleViewSet(viewsets.ReadOnlyModelViewSet):
...
+        queryset = Exercise.objects.all()
+        filter_value = self.request.query_params.get('id', None)
+        if filter_value is not None:
+            queryset = queryset.filter(field_name=filter_value)
+        return queryset
+
```
Comment (frankip) [inline wger/exercises/api/views.py]: add a new line at the end to follow the PEP8 standards

File: wger/urls.py [hunk 9]
```diff
@@ -169,6 +172,11 @@
         nutrition_api_views.search,
         name='ingredient-search'),
     url(r'^api/v2/', include(router.urls)),
+
+    # Register view with the url
```
Comment (frankip) [inline wger/urls.py]: nice work

**Review by frankip — COMMENTED** [review]
great job 👍

**Review by mutoromike — COMMENTED** [review]
Good work Felix. I like how you have implemented your classes and also comments. Thumbs up for including the dotfiles too.

**Review by mutoromike — APPROVED** [review]

**Pull request merged**